cpp/src/arrow/csv/writer.cc

```diff
@@ -80,15 +80,19 @@ int64_t CountQuotes(util::string_view s) {
 // Matching quote pair character length.
 constexpr int64_t kQuoteCount = 2;
 constexpr int64_t kQuoteDelimiterCount = kQuoteCount + /*end_char*/ 1;
+constexpr char const* kStrComma = ",";
```

Comment:
```suggestion
constexpr const char * kStrComma = ",";
```

See: https://medium.com/@samathy_barratt/constantly-confusing-c-const-and-constexpr-pointer-behaviour-7eddb041e15a